src/main/java/uk/gov/hmcts/reform/managecase/service/CaseAssignmentService.java

```diff
@@ -138,13 +144,25 @@ private CaseDetails getCase(CaseAssignment input) {
             .collect(toList());
     }
 
-    private Optional<ProfessionalUser> findUserBy(List<ProfessionalUser> users, String assigneeId) {
-        return users.stream()
-                .filter(user -> assigneeId.equalsIgnoreCase(user.getUserIdentifier()))
-                .findFirst();
+    private boolean isAssigneePresent(List<ProfessionalUser> users, String assigneeId) {
+        return users.stream().anyMatch(user -> assigneeId.equalsIgnoreCase(user.getUserIdentifier()));
     }
 
     private boolean containsIgnoreCase(List<String> list, String value) {
         return list.stream().anyMatch(value::equalsIgnoreCase);
     }
+
+    private List<String> getAssigneeRoles(String assigneeId) {
+        String systemUserToken = idamRepository.getSystemUserAccessToken();
+        List<uk.gov.hmcts.reform.idam.client.models.UserDetails> users =
+                idamRepository.searchUsers(systemUserToken, String.format(IDAM_ES_QUERY, assigneeId));
```

Comment:
We could just assume 1 and only 1 user to be returned from IDAM, and write the code accordingly. Also, it would be better to keep the service class illiterate the token and query thing. The repository would be better to just take the user id and return the user object. In this situation, the data retrieval logic is contaminated into business logic.

Reply:
I thought we might need multi users search very soon... so leave that adjustment to caller (service class). 
But I agree it would be nice to abstract in repo as much as possible. Will rename repo method as
 `UserDetail searchUser(String userId)`. I think caller still need to pass token because of Spring proxying used cache annotation.  Cache will not work on a method if it invoked from another method in the class.

Reply:
We can have 2 methods, 1 taking a user id and returning a user object, other taking a variable list of user ids and returning a list of user objects. Each may have mostly similar or somewhat different implementations, but both to abstract out the details like even the ES query. The invoker of the repo method doesn't need to know about the fact that an ES-query is used to actually get the users. Now, we can use one of the 2 methods. When multi user access is needed, we can use the other one. In each case, service class stays to the very minimum of impl details pf repo class.

Reply:
Yeah.. sounds good. Pushed the change for one user scenario.

Reply:
Thanks Sateesh.